### PR TITLE
Added testing infrastructure

### PR DESCRIPTION
**Pull request by lucacasonato**
Had to move `react-syntax-highlighter` from `esm` to `cjs` dist because `esm` is not supported in node, where the tests run.

Progress towards #44

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,9 +12,7 @@ jobs:
           node-version: 12
       - run: yarn --frozen-lockfile
       - run: yarn run lint
-      #
-      # TODO(ry) tests!
-      #
+      - run: CI=true yarn test
```
Comment (ry) [inline .github/workflows/build.yml]: 😀

**Comment by ry** [comment]
@lucacasonato It would be helpful if you allowed me to push to your branches - otherwise if I want to make change I have to start a new PR.

**Comment by lucacasonato** [comment]
> @lucacasonato It would be helpful if you allowed me to push to your branches - otherwise if I want to make change I have to start a new PR.

I have that enabled.
image

**Comment by ry** [comment]
@lucacasonato Oh sorry, it must be something wrong with my setup. 


I'm trying to modify the test ... do you have a suggestion of how to make this work? https://gist.github.com/ry/3fe4812078bb7ec33cb1e78d88db1516

**Comment by ry** [comment]
( I think it's because you're sending a branch from a different user @luca-rand )

**Comment by lucacasonato** [comment]
Invited you: https://github.com/luca-rand/deno_website2/invitations - you should be able to commit then

**Comment by lucacasonato** [comment]
> I'm trying to modify the test ... do you have a suggestion of how to make this work? https://gist.github.com/ry/3fe4812078bb7ec33cb1e78d88db1516

You seem to not import `render`: `import { render } from "@testing-library/react";`

**Comment by ry** [comment]
@lucacasonato Thank you - I'm quite ignorant of React.

That test seems to work now, but getting a strange warning...
```
  console.error node_modules/react-dom/cjs/react-dom.development.js:530
    Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
    
    Check the render method of `ForwardRef(Typography)`.
        in HashLink (created by ForwardRef(Typography))
        in ForwardRef(Typography) (created by WithStyles(ForwardRef(Typography)))
        in WithStyles(ForwardRef(Typography)) (created by ForwardRef(Link))
        in ForwardRef(Link) (created by WithStyles(ForwardRef(Link)))
        in WithStyles(ForwardRef(Link)) (at Home.js:153)
        in b (at Home.js:152)
        in p (at Home.js:151)
        in main (at Home.js:19)
        in Home (at Home.test.tsx:7)
        in Router (created by MemoryRouter)
        in MemoryRouter
```

**Comment by lucacasonato** [comment]
@ry Found the issue. It seems to be an incompatibility between `material-ui` and `react-router-hash-link`. I have a solution for it - will open a PR in a minute.

**Pull request merged**